src/Microsoft.Health.Fhir.Shared.Core/Extensions/ModelExtensions.cs

```diff
@@ -115,7 +115,8 @@ public static SearchParameterInfo ToInfo(this SearchParameter searchParam)
 
             return new SearchParameterInfo(
                 searchParam.Name,
-                searchParam.Type?.ToString(),
+                searchParam.Code,
+                Enum.Parse<ValueSets.SearchParamType>(searchParam.Type?.ToString()),
```

Comment:
>searchParam.Type [](start = 54, length = 16)

Wouldn't `searchParam.Type` be enough here?

Reply:
Unfortunately, searchParam.Type is the class from the firely library, and ValueSets.SearchParamType is one of our own classes, so we need to Parse it to convert it.